app/models/interval_receiver.rb

```diff
@@ -1,10 +1,20 @@
 # frozen_string_literal: true
 
 class IntervalReceiver < EventReceiver
+  # Callbacks
+  after_create :create_internal_job!
+
+  # Validations
   validates :start_time, :interval, presence: true
 
-  # Only called from sidekiq job, at certain intervals
+  # Sidekiq job scheduled after_create => trigger condition will always be met
   def trigger_condition_met?(_job)
     true
   end
+
+  private
+
+  def create_internal_job!
+    InternalWorker.perform_at(interval.from_now, id)
```

Comment:
So we dont actually want from_now. We want `start_time + n*interval.seconds`, where `n` is the smallest such that the time is in the future